### PR TITLE
Fix Kitty wrapper on MacOS

### LINKED ISSUE #61231: Kitty wrapper does not work on MacOS
## Issue description
Due to MacOS oddities `kitty` does not work when executed as symlink(https://github.com/kovidgoyal/kitty/issues/1539) and hence, the wrapper installed via https://github.com/NixOS/nixpkgs/blob/48a65ef9e62142712ca7e91d472787554a32649b/pkgs/applications/misc/kitty/default.nix#L93
cannot work on MacOS.
In fact, `${pkgs.kitty}/bin/kitty` is unusable on MacOS whatever the content is.

Any idea what could be done about this?
Should we remove the `bin` directory altogether from the build output to prevent confusion?

### PR DESCRIPTION
**Pull request by rvolosatovs**
###### Motivation for this change
Closes https://github.com/NixOS/nixpkgs/issues/61231

Apply https://github.com/kovidgoyal/kitty/commit/bdeec612667f6976109247fe1750b10dda9c24c0 and https://github.com/kovidgoyal/kitty/commit/af2c9a49b1ad31e94242295d88598591623fbf11 as a patch

Refs https://github.com/kovidgoyal/kitty/issues/1539

The patch can be removed once the changes referenced land in a release

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/kitty/default.nix [hunk 1]
```diff
@@ -68,6 +68,7 @@ buildPythonApplication rec {
     })
   ] ++ stdenv.lib.optionals stdenv.isDarwin [
     ./macos-10.11.patch
+    ./macos-symlink.patch
```
Comment (veprbl) [inline pkgs/applications/misc/kitty/default.nix]: Can you use `fetchpatch` for this?

Comment (veprbl) [reply pkgs/applications/misc/kitty/default.nix]: https://github.com/kovidgoyal/kitty/commit/bdeec612667f6976109247fe1750b10dda9c24c0.patch

**Comment by rvolosatovs** [comment]
@veprbl

**Review by veprbl — APPROVED** [review]

**Comment by veprbl** [comment]
@rvolosatovs Thank you!

**Pull request merged**